oryx-app-mllib/src/main/java/com/cloudera/oryx/app/mllib/kmeans/KMeansEvaluation.java

```diff
@@ -122,10 +124,48 @@
 
     double dunnIndex = minInterClusterDistance / maxIntraClusterDistance;
     log.info("Computed Dunn Index for {} clusters: {}", numClusters, dunnIndex);
-
     return dunnIndex;
   }
 
+  /**
+   * Computes the Silhouette Coefficient for this clustering, range is [-1,1]
+   * (http://www.sciencedirect.com/science/article/pii/0377042787901257)
+   * Here we are computing the overall Silhouette Coefficient across all of the clustering
+   * Other option would be to calculate the Silhouette coefficient for each of the individual clusters
+   * @param evalData data for evaluation
+   * @return Silhouette Coefficient, closer to 1 is better
+   */
+  double silhouetteCoefficient(JavaRDD<Vector> evalData) {
+    JavaRDD<Vector> downSampleData = evalData.sample(false, 0.2, RandomManager.getRandom().nextLong());
```

Comment:
I think we can not (yet) make the sampling rate configurable. But rather than fix a percentage, let's maybe fix an arbitrary max size. For example, 100K elements? then you only down-sample if it's larger than that, and by a factor of 100K / rdd.size